Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -37,6 +37,14 @@ public struct IndexedRouteResponse {
     }
 }
 
+/**
+ A closure to be called when `RouteLeg` was changed.
+ 
+ - parameter success: Boolean value, which is set to `true` in case if `RouteLeg` was
+ successfully changed, otherwise `false`.
+ */
+public typealias AdvanceLegCompletionHandler = (_ success: Bool) -> Void
```

Comment:
This is a breaking change, which affects both `RouteController` and `LegacyRouteController`. I'm also hesitant about introducing it: previously we did not provide any information to the user about success. Now it's available, which might be useful.

Reply:
Similar to https://github.com/mapbox/mapbox-navigation-ios/pull/3342#discussion_r709664302, it would be helpful to define this completion handler to accept a `Result<Error, RouteProgress>`. Even though the navigator doesn’t provide us with an Error in the failure case, we can create a basic one on the spot. In the future, if the navigator does start providing more information, we can naturally pass it along without having to break the public API. But if we don’t go through that trouble now, we can introduce a parallel method and completion handler type when the time comes.

Remember to mention the change in the changelog.

Reply:
`UpdateLocationCompletionHandler` and `AdvanceLegCompletionHandler` were modified to use `Result` instead of `Boolean`. Error enums are internal, which gives us an opportunity to change them in future.